core/injector/block_producing_node_injector.hpp

```diff
@@ -29,28 +29,31 @@ namespace kagome::injector {
       Ts &&... args) {
     using namespace boost;  // NOLINT;
 
-    auto get_babe_observer = get_babe;
-
     return di::make_injector(
 
         // inherit application injector
         makeApplicationInjector(
             genesis_path, leveldb_path, rpc_http_endpoint, rpc_ws_endpoint),
         // bind sr25519 keypair
-        di::bind<crypto::SR25519Keypair>.to(std::move(get_sr25519_keypair)),
+        di::bind<crypto::SR25519Keypair>.to(
+            [](auto const &inj) { return get_sr25519_keypair(inj); }),
         // bind ed25519 keypair
-        di::bind<crypto::ED25519Keypair>.to(std::move(get_ed25519_keypair)),
+        di::bind<crypto::ED25519Keypair>.to(
+            [](auto const &inj) { return get_ed25519_keypair(inj); }),
```

Comment:
Do we need such changes to be compatible with gcc10?

Reply:
We need to remove lambda objects as a global variables in headers.